.travis.yml

```diff
@@ -46,15 +57,61 @@ matrix:
       env: [FRONTEND=2.068]
     - d: gdc-4.8.5
       env: [FRONTEND=2.068]
-
-  allow_failures:
-    - d: gdc
-
-addons:
-  apt:
-    packages:
-      - libevent-dev
-
-script:
-  - deactivate # deactivate host compiler
-  - ./travis-ci.sh
+    - stage: deploy
+      d: ldc
```

Comment:
I wasn't sure how the DUB binaries are currently generated, but we probably want to use LDC and it's amazing optimizer.
In a next step, LTO + PGO should/could follow.

BTW for reference:

Compiler | Size
-------|------
DMD | 17M
LDC | 13M
GDC | 19M

(though of course size isn't all, but I don't have the time to run a benchmark at the moment, but I strongly assume that `ldc` is our best choice)

Reply:
Regarding binary size, I'd keep debug info, though, because it often considerably speeds up the diagnostic process in case of bugs. BTW, does LDC generate stack traces with names for uncaught exceptions? I remember that this didn't use to be the case.

Reply:
> Regarding binary size, I'd keep debug info, though, because it often considerably speeds up the diagnostic process in case of bugs.

Sure, `build.sh` already adds `-g` and I never argued against ;-)

> BTW, does LDC generate stack traces with names for uncaught exceptions? I remember that this didn't use to be the case.

https://run.dlang.io/is/zsSvOA (nope it does't)

If that's a deal breaker, we can also switch to `dmd`. It's a very simple change in the `travis.yml` config after all.

Reply:
Shared or static linking? :-)

Reply:
With static linking on Linux (at least for DMD + LDC).